### PR TITLE
[DON'T MERGE] chore(datasets): Update setup.py for snowpark

### PR DESCRIPTION
**Pull request by noklam**
## Description

The current pin for `snowflake-snowpark-python` is `1.0.0`, which only support Python 3.8, the latest version support Python 3.8 & 3.9 instead.

https://pypi.org/project/snowflake-snowpark-python/#history
## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by noklam** [comment]
Pinging @heber-urdaneta @mobilize-mrojas @orellabac @Vladimir-Filimonov as I found you are the original author of this PR https://github.com/kedro-org/kedro-plugins/pull/104

**Review by astrojuanlu — APPROVED** [review]

**Comment by Vladimir-Filimonov** [comment]
Hey all. I tried running snowflake unit tests using this PR and virtual environment of python 3.9 - no luck, requirements installation fails. If I try to repeat CI steps from here https://github.com/kedro-org/kedro-plugins/blob/main/.github/workflows/check-plugin.yml#L40
installation of requirements fails:
- version in test_requirements need to be updated accordingly: https://github.com/kedro-org/kedro-plugins/blob/fix-snowpark/kedro-datasets/test_requirements.txt#L54
- once I updated it and made it look like `snowflake-snowpark-python~=1.0` - installation fails complaining missing class from arrow: "Package 'snowflake.connector.cpp.ArrowIterator' is absent from the `packages` configuration.". I would guess its because of requirements referencing pyarrow this way https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/test_requirements.txt#L39 (it was part of the original PR for snowflake connector, as at that times connector required this version of pyarrow). 

I'm on vacation now and don't have much time to experiment myself, but if you can make requirements installation work I'm happy to run [Snowflake unit tests](https://github.com/kedro-org/kedro-plugins/tree/fix-snowpark/kedro-datasets/tests/snowflake) to confirm everything is alright. 


Other suggestion:
- update or remove this line from README: https://github.com/kedro-org/kedro-plugins/blob/fix-snowpark/kedro-datasets/tests/snowflake/README.md?plain=1#L23

**Comment by noklam** [comment]
We are working on adding python 3.11 support and it's likely we will bump many of the dependency to support Spark better. Thanks for looking into it @Vladimir-Filimonov

**Comment by astrojuanlu** [comment]
Now that #297 is merged, maybe this is worth revisiting?

**Comment by noklam** [comment]
I checked again, they now support >=3.8 <3.11. Last time I try to ask the author of SnowparkDataset to test with Python 3.9 and it breaks.

I think the challenge here is how to we maintain this dataset as we don't have any test suite to run against Snowflake.

File: kedro-datasets/setup.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def _collect_requirements(requires):
     "requests-mock~=1.6",
     "requests~=2.20",
     "s3fs>=0.3.0, <0.5",  # Needs to be at least 0.3.0 to make use of `cachable` attribute on S3FileSystem.
-    "snowflake-snowpark-python~=1.0.0; python_version == '3.8'",
+    "snowflake-snowpark-python~=1.0.0; python_version < '3.11'",
```
Comment (astrojuanlu) [inline kedro-datasets/setup.py:205]:
```suggestion
    "snowflake-snowpark-python~=1.0; python_version < '3.11'",
```

?

**Comment by astrojuanlu** [comment]
This was done in #442

**Pull request closed without merging**